Lib/fractions.py

```diff
@@ -798,17 +806,17 @@ def __pow__(a, b):
             if b.denominator == 1:
                 power = b.numerator
                 if power >= 0:
-                    return Fraction(a._numerator ** power,
-                                    a._denominator ** power,
-                                    _normalize=False)
-                elif a._numerator >= 0:
-                    return Fraction(a._denominator ** -power,
-                                    a._numerator ** -power,
-                                    _normalize=False)
+                    return Fraction._from_pair(a._numerator ** power,
+                                               a._denominator ** power)
+                elif a._numerator > 0:
+                    return Fraction._from_pair(a._denominator ** -power,
+                                               a._numerator ** -power)
+                elif a._numerator == 0:
+                    raise ZeroDivisionError('Fraction(%s, 0)' %
```

Comment:
```suggestion
                    raise ZeroDivisionError('Fraction(0, %s)' %
```

Reply:
I would like to reproduce the current exception, e.g.:
```
>>> fractions.Fraction(0)**-1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sk/src/cpython/Lib/fractions.py", line 805, in __pow__
    return Fraction(a._denominator ** -power,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sk/src/cpython/Lib/fractions.py", line 281, in __new__
    raise ZeroDivisionError('Fraction(%s, 0)' % numerator)
ZeroDivisionError: Fraction(1, 0)
```

Reply:
I understand. I find the original behaviour a bit strange, as `Fraction(5,1)/Fraction(0)` and `Fraction(0)**-2` also result in `ZeroDivisionError: Fraction(1, 0)`. Where did the `Fraction(1,0)` originate from? This is something internal, not something the user entered.

We could modify the error message a bit so that `Fraction(5,1)/Fraction(0)` results in 
```
`ZeroDivisionError: division by Fraction(0, 1)`
```
That is more in line with the behaviour for `int` or `float`. E.g. `2/0` results in 
```
ZeroDivisionError: division by zero
```

But if this PR just reproduces the original behaviour that is fine and we could create a separate issue for this.

Reply:
> a bit strange

It says: "You are trying to create Fraction(foo, 0)".  Looks reasonable.

> Where did the Fraction(1,0) originate from?

I think, it's from 7736b5becd...

Reply:
Given that this PR is supposed to be optimization-only, I agree with @skirpichev that the exception messages should stay unchanged. @eendebakpt If you feel strongly enough about the error message, let's open a separate issue for that.

Reply:
I agree that the issues are better handled in a separate pr. Given the fact that this error message has been around for a very long time, i am not sure we should change it.